### PR TITLE
feat(mq): add new check `mq_broker_auto_minor_version_upgrades`

### PR DESCRIPTION
**Pull request by HugoPBrito**
### Context

`Amazon MQ` is a managed message broker service that simplifies the setup and maintenance of message brokers for applications. It supports multiple messaging protocols and is designed to work with various broker engines like `ActiveMQ` and `RabbitMQ`.

Keeping the broker engines up to date is crucial for maintaining security and performance, especially as vulnerabilities and bugs are discovered over time.

### Description

This check verifies whether Amazon `MQ brokers` have `automatic minor version upgrades` enabled. The check fails if this feature is not activated, which could expose the broker to security vulnerabilities and performance issues.

### Checklist

- Are there new checks included in this PR? Yes.
    - If so, do we need to update permissions for the provider? No.
- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [x] Review if backport is needed.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/mq/mq_service.py [hunk 1]
```diff
@@ -39,3 +55,5 @@ class Broker(BaseModel):
     name: str
     id: str
     region: str
+    auto_minor_version_upgrade: bool = False
+    tags: Optional[list] = []
```
Comment (puchy22) [inline prowler/providers/aws/services/mq/mq_service.py]: If you are going to assign tags in the check, please ensure to attach it first in the describe call

File: tests/providers/aws/services/mq/mq_service_test.py [hunk 2]
```diff
@@ -64,3 +64,39 @@ def test_list_brokers(self):
         assert mq.brokers[broker_arn].name == "my-broker"
         assert mq.brokers[broker_arn].region == AWS_REGION_EU_WEST_1
         assert mq.brokers[broker_arn].id == broker["BrokerId"]
+
+    # Test MQ Describe Broker
+    @mock_aws
+    def test_describe_broker(self):
+        # Generate MQ client
```
Comment (puchy22) [inline tests/providers/aws/services/mq/mq_service_test.py]:
```suggestion
```

File: tests/providers/aws/services/mq/mq_service_test.py [hunk 3]
```diff
@@ -64,3 +64,39 @@ def test_list_brokers(self):
...
+                    "Groups": [],
+                    "Password": "password",
+                    "Username": "user",
+                }
+            ],
+        )
+        broker_arn = broker["BrokerArn"]
+        broker["BrokerId"]
+
+        # MQ Client for this test class
```
Comment (puchy22) [inline tests/providers/aws/services/mq/mq_service_test.py]:
```suggestion
```

File: tests/providers/aws/services/mq/mq_broker_auto_minor_version_upgrades/mq_broker_auto_minor_version_upgrades_test.py [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        aws_provider = set_mocked_aws_provider([AWS_REGION_US_EAST_1])
+
+        with mock.patch(
+            "prowler.providers.common.provider.Provider.get_global_provider",
+            return_value=aws_provider,
+        ):
+            with mock.patch(
+                "prowler.providers.aws.services.mq.mq_broker_auto_minor_version_upgrades.mq_broker_auto_minor_version_upgrades.mq_client",
+                new=MQ(aws_provider),
+            ):
```
Comment (puchy22) [inline tests/providers/aws/services/mq/mq_broker_auto_minor_version_upgrades/mq_broker_auto_minor_version_upgrades_test.py]:
```suggestion
        with mock.patch(
            "prowler.providers.common.provider.Provider.get_global_provider",
            return_value=aws_provider,
        ), mock.patch(
           "prowler.providers.aws.services.mq.mq_broker_auto_minor_version_upgrades.mq_broker_auto_minor_version_upgrades.mq_client",
                new=MQ(aws_provider),
            ):
```

File: tests/providers/aws/services/mq/mq_broker_auto_minor_version_upgrades/mq_broker_auto_minor_version_upgrades_test.py [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
```
Comment (puchy22) [inline tests/providers/aws/services/mq/mq_broker_auto_minor_version_upgrades/mq_broker_auto_minor_version_upgrades_test.py]:
```suggestion
        with mock.patch(
            "prowler.providers.common.provider.Provider.get_global_provider",
            return_value=aws_provider,
        ), mock.patch(
                "prowler.providers.aws.services.mq.mq_broker_auto_minor_version_upgrades.mq_broker_auto_minor_version_upgrades.mq_client",
                new=MQ(aws_provider),
            ):
```

File: tests/providers/aws/services/mq/mq_broker_auto_minor_version_upgrades/mq_broker_auto_minor_version_upgrades_test.py [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+                # Test Check
+                from prowler.providers.aws.services.mq.mq_broker_auto_minor_version_upgrades.mq_broker_auto_minor_version_upgrades import (
+                    mq_broker_auto_minor_version_upgrades,
+                )
```
Comment (puchy22) [inline tests/providers/aws/services/mq/mq_broker_auto_minor_version_upgrades/mq_broker_auto_minor_version_upgrades_test.py]:
```suggestion
        with mock.patch(
            "prowler.providers.common.provider.Provider.get_global_provider",
            return_value=aws_provider,
        ), mock.patch(
                "prowler.providers.aws.services.mq.mq_broker_auto_minor_version_upgrades.mq_broker_auto_minor_version_upgrades.mq_client",
                new=MQ(aws_provider),
            ):
                # Test Check
                from prowler.providers.aws.services.mq.mq_broker_auto_minor_version_upgrades.mq_broker_auto_minor_version_upgrades import (
                    mq_broker_auto_minor_version_upgrades,
                )
```

**Review by puchy22 — DISMISSED** [review]

**Review by MrCloudSec — APPROVED** [review]

**Pull request merged**